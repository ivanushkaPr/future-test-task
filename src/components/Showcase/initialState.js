const initialState = {
  "Fashion": {
    goods: [
      
    ],
  }, 
  "Furnitures": {
    goods: [
      
    ],
  }, 
  "Watches": {
    goods: [
      {name: "Fashion Genuine Chain Watch", id: 1, price: "25.36"},
      {name: "Fashion Silver Chain Watch", id: 2, price: "35.24"}, 
      {name: "Fashion Chain Watch", id: 3, price: "23.12"}, 
      {name: "Fashion Black Chain Watch", id: 4, price: "42.32"}, 
      {name: "Fashion Lether Watch", id: 5, price: "15.20"}, 
      {name: "Fashion Forign Watch", id: 6, price: "36.20"}, 
    ],
    folderName: 'watches/',
    prefix: "watch"
  }, 
  "Mobile": {
    goods: [
      {name: "Fast phone", id: 1, price: "25.36"},
      {name: "Cheap Phone", id: 2, price: "35.24"}, 
      {name: "Fast phone", id: 3, price: "23.12"}, 
      {name: "Cheap Phone", id: 4, price: "42.32"}, 
      {name: "Fast phone", id: 5, price: "15.20"}, 
      {name: "Cheap Phone", id: 6, price: "36.20"}, 
    ],
    folderName: 'mobile/',
    prefix: "mobile"
  }, 
  "Cloth": {
    goods: [
      {name: "Hood", id: 1, price: "25.36"},
      {name: "Polo", id: 2, price: "35.24"}, 
      {name: "Polo", id: 3, price: "23.12"}, 
      {name: "Polo", id: 4, price: "42.32"}, 
      {name: "Polo", id: 5, price: "15.20"}, 
      {name: "Polo", id: 6, price: "36.20"}, 
    ],
    folderName: 'cloth/',
    prefix: "cloth"
  }, 
  "Toys": {
    goods: [
      
    ],
  },
  active:  "Watches",
};

export default initialState;