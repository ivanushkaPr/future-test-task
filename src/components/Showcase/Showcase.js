import React, { useState } from 'react';
import styled from 'styled-components';
import Categories from './Categories/Categories'
import ProductCard from './ProductCard/ProductCard';
import initialState from './initialState';
import circle from './images/circle.png';
import square from './images/square.png';
import triangle from './images/bigTriangle.png';
import transTriangle from './images/transTriangle.png';
import line from './images/line.png'


const Container = styled.div`
background-image: url(${triangle}), url(${line});
background-repeat: no-repeat;
background-position: 163px 38px, 801px 52px;
margin-top: 84px;
background-color: white;
position: static;
margin-left: 123px;
margin-right: 95px;
padding-bottom: 65px;
border-top: 0.1px solid transparent;
box-shadow: 0px 5px 47.5px 2.5px rgba(237, 235, 253, 0.75);
align-content: stretch;
`;

const Figure = styled.img`
position: absolute;
background-image: url(${props => props.src});
top: ${props => props.top};
left: ${props => props.left};
`;

const Products = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: stretch;
  margin-left: 107px;
  margin-right: 109px;
`;

const Title = styled.h5`
margin-top: 100px;
font-family: OpenSans-Bold;
font-size: 50px;
line-height: 26px;
`;

const Showcase = props => {
  const [state, setState] = useState(initialState);

  const active = state.active;
  const renderData = Object.keys(state).slice(0, Object.keys(state).length - 1);

  const match = renderData.find(category => category === active);
  

  const onClickCategoryHandler = e => {
    const category = e.currentTarget.dataset.category;
    
    setState({
      ...state,
      active: category
    })
  }

  return (
    <Container>
        <Title> Our Showcase </Title>
        <Figure src={circle} left={'257px'} top={'256px'}/>
        <Figure src={transTriangle} left={'65px'} top={'868px'}/>
        <Figure src={square} left={'1485px'} top={'477px'}/>
      <Categories click={onClickCategoryHandler} data={state}/>
      <Products>
        {
          state[match].goods.map(good => {
            const src = require('./images/' + state[match].folderName + state[match].prefix + good.id +'.png')
            console.log(src)
            return (
              <ProductCard key={src} data={good} src={src}/>
            )
          })

        }
      </Products>
    </Container>
  );
}

export default Showcase;