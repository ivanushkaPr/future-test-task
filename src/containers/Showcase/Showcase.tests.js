import React from 'react';
import ReactDOM from 'react-dom';
import Showcase from './Showcase';
import renderer from 'react-test-renderer';

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Showcase/>, div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Showcase/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });






