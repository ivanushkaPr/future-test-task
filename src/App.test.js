import React from 'react';
import { render } from '@testing-library/react';

import './containers/Showcase/Showcase.tests'