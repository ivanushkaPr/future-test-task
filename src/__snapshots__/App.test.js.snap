// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxmLO hvfItH"
>
  <h5
    className="sc-fzpans iayjxY"
  >
     Our Showcase 
  </h5>
  <img
    className="sc-fzozJi kNkiiB"
    src="circle.png"
  />
  <img
    className="sc-fzozJi boqogr"
    src="transTriangle.png"
  />
  <img
    className="sc-fzozJi fkFyoD"
    src="square.png"
  />
  <nav
    className="sc-AxjAm fTPNHu"
  >
    <ul
      className="sc-AxirZ feCToI"
    >
      <li
        className="sc-AxiKw bOTXee"
        data-category="Fashion"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Fashion
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Furnitures"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Furnitures
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Watches"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb lpfXDb"
        >
          Watches
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Mobile"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Mobile
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Cloth"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Cloth
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Toys"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Toys
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="sc-fzoLsD fGoMsa"
  >
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch1.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Genuine Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        25.36
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch2.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Silver Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        35.24
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch3.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        23.12
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch4.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Black Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        42.32
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch5.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Lether Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        15.20
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch6.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Forign Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        36.20
         
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="sc-AxmLO hvfItH"
>
  <h5
    className="sc-fzpans iayjxY"
  >
     Our Showcase 
  </h5>
  <img
    className="sc-fzozJi kNkiiB"
    src="circle.png"
  />
  <img
    className="sc-fzozJi boqogr"
    src="transTriangle.png"
  />
  <img
    className="sc-fzozJi fkFyoD"
    src="square.png"
  />
  <nav
    className="sc-AxjAm fTPNHu"
  >
    <ul
      className="sc-AxirZ feCToI"
    >
      <li
        className="sc-AxiKw bOTXee"
        data-category="Fashion"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Fashion
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Furnitures"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Furnitures
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Watches"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb lpfXDb"
        >
          Watches
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Mobile"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Mobile
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Cloth"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Cloth
        </a>
      </li>
      <li
        className="sc-AxiKw bOTXee"
        data-category="Toys"
        onClick={[Function]}
      >
        <a
          className="sc-AxhCb kSSttd"
        >
          Toys
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="sc-fzoLsD fGoMsa"
  >
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch1.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Genuine Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        25.36
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch2.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Silver Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        35.24
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch3.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        23.12
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch4.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Black Chain Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        42.32
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch5.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Lether Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        15.20
         
      </p>
    </div>
    <div
      className="sc-AxhUy bkLdaw"
    >
      <img
        className="sc-AxgMl igILJB"
        src="watch6.png"
      />
      <p
        className="sc-AxheI dWsJZJ"
      >
         
        Fashion Forign Watch
         
      </p>
      <p
        className="sc-Axmtr kciBKh"
      >
         $ 
        36.20
         
      </p>
    </div>
  </div>
</div>
`;
